File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.py
class Solution:
       def countSubarrays(self, A, k):
        res = cur = i = 0
        for j in range(len(A)):
            cur += A[j]
            while cur * (j - i + 1) >= k:
                cur -= A[i]
                i += 1
            res += j - i + 1
        return res

        

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.cpp
class Solution {
public:
   long long countSubarrays(vector<int>& A, long long k) {
        long long res = 0, cur = 0;
        for (int j = 0, i = 0; j < A.size(); ++j) {
            cur += A[j];
            while (cur * (j - i + 1) >= k)
                cur -= A[i++];
            res += j - i + 1;
        }
        return res;
    }
};